### ISSUE TITLE
loading additional services via GTM

### ISSUE THREAD

**Issue by jshwlkr** [umdigital/umich-cookie-consent#3]
I'm currently trying to parse through OneTrust's documentation for GTM, but I'm assuming (even if I'm using SiteKit to load GTM) I have to check for consent on GTM's end somehow before loading any additional 3rd-party service. Not quite sure if there's a preferred method for doing that in this specific case.

**Comment by nobleclem** [comment]
I have never actually used GTM for loading 3rd party assets, however I think you should be able to edit the 3rd party tag and in Advanced Settings there is a "Consent Settings" option.  From there you can select "Require additional consent for tag to fire" and can select the consent setting.  Most likely this will be "ad_storage" or "analytics_storage".  In theory I think that would be enough to determine when that tag is inserted based on consent.  I found this [article](https://www.simoahava.com/analytics/consent-settings-google-tag-manager/) about the consent signals which are set.  I skimmed through it and it seems like it has better information.

Let me know if it looks like that works or not.  I will have to run some tests myself as time permits.

**Comment by nobleclem** [comment]
I was able to do some testing with this.  The solution is not perfect but it should work.  I have updated the plugin to include a new event "um_consent_updated" that can be used as a trigger in GTM.  Doing that in combination with the Advanced Settings -> Consent Settings in GTM for a tag that seems to work in my tests.  The caveat would be if you had a button available to allow someone to change their settings and they were to toggle it off and on then the GTM Tag would be executed on each allow.  I currently have not found a solution to resolve that but that should be close 0 for real world case.
